localization/yabloc/yabloc_pose_initializer/src/camera/camera_pose_initializer_core.cpp

```diff
@@ -110,9 +106,21 @@ bool CameraPoseInitializer::estimate_pose(
     using namespace std::literals::chrono_literals;
     std::future_status status = result_future.wait_for(1000ms);
     if (status == std::future_status::ready) {
-      segmented_image = result_future.get()->dst_image;
+      const auto response = result_future.get();
+      if (response->success) {
+        segmented_image = response->dst_image;
+      } else {
+        RCLCPP_WARN_STREAM(get_logger(), "segmentation service failed unexpectedly");
+        // NOTE: Even if the segmentation service fails, the function will still return the
+        // yaw_angle_rad as it is and complete the initialization. The service fails
+        // when the DNN model is not downloaded. Ideally, initialization should rely on
+        // segmentation, but this implementation allows initialization even in cases where network
+        // connectivity is not available.
+        return true;
```

Comment:
Now the functions seems a bit overcomplicated, how about this?

Make the arguments of this function from
```
bool CameraPoseInitializer::estimate_pose(
  const Eigen::Vector3f & position, double & yaw_angle_rad, double yaw_std_rad)
```
to
```
bool CameraPoseInitializer::estimate_pose(
  const Eigen::Vector3f & position,
  const double yaw_angle_rad_input,
  const double yaw_std_rad_input,
  double & yaw_angle_rad)
```

and then
```suggestion
        RCLCPP_WARN_STREAM(get_logger(), "segmentation service failed unexpectedly");
        // NOTE: Even if the segmentation service fails, the function will still return the
        // yaw_angle_rad as it is and complete the initialization. The service fails
        // when the DNN model is not downloaded. Ideally, initialization should rely on
        // segmentation, but this implementation allows initialization even in cases where network
        // connectivity is not available.
        yaw_angle_rad = yaw_angle_rad_input;
        return true;
```

Reply:
Thanks for the insightful advice. 
I modified it to return std::optional(yaw_angle_rad) instead of a reference type argument.
How about now?  1701912

Reply:
Perfect!